src/rust/ide/view/graph-editor/src/component/node/output/port.rs

```diff
@@ -436,6 +504,17 @@ impl Model {
 
             // === Size ===
 
+            frp.source.size <+ frp.set_size;
+            eval frp.size ((&s)
+                shape.set_size(s + Vector2(HOVER_AREA_PADDING,HOVER_AREA_PADDING) * 2.0));
+            set_type_label_x <- all_with(&frp.size,&type_label.width,
+                f!([port_count,port_index](port_size,type_label_width) {
+                    let shape_border_length = shape_border_length(node::RADIUS, port_size.x);
+                    let relative_x          = (port_index as f32 + 0.5) / (port_count as f32) - 0.5;
+                    let center_x            = relative_x * shape_border_length;
```

Comment:
What are these variables used for? The names `relative_x` and `center_x` do not explain what they are used for. What is `relative_x` relative to? Could you please improve the names so their meaning would be obvious?

Also, the logic is strange IMO. Why are you adding and removing `0.5`? For example, what if `port_index` is `1` and `port_count` is `2`? Then your equation is `(1 + 0.5) / 2 - 0.5 = 1.5 / 2.0 - 0.5 = 0.25` - TBH I don't understand reasoning behind it. This should be explained much better.